src/util/markdownToText.ts

```diff
@@ -0,0 +1,240 @@
+import { Token, Tokens, marked } from 'marked'
+
+/** Converts markdown to text compatible with `importText`. */
+export const markdownToText = (markdown: string): string => {
+  const tokens = marked.lexer(markdown.trim())
+
+  // Accumulate the result
+  let result = '\n'
+
+  /** Converts and combines inline tokens to a string. */
+  const processInlineTokens = (tokens: Token[] | string) => {
+    // Convenience for allowing to pass in a string directly, allows skipping ternaries when calling recursively.
+    if (typeof tokens === 'string') return tokens
+
+    let str = ''
+
+    for (const token of tokens) {
+      switch (token.type) {
+        case 'text':
+          str += token.text.replace(/\n/g, '&#10;')
+          break
+        case 'em':
+          str += `<i>${processInlineTokens(token.tokens ?? token.text)}</i>`
+          break
+        case 'strong':
+          str += `<b>${processInlineTokens(token.tokens ?? token.text)}</b>`
+          break
+        case 'del':
+          str += `<strike>${processInlineTokens(token.tokens ?? token.text)}</strike>`
+          break
+        case 'link':
+          if (token.href === token.text) str += token.text
+          else str += `<a href="${token.href}">${processInlineTokens(token.tokens ?? token.text)}</a>`
+          break
+        case 'image':
+          str += `<img src="${token.href}" alt="${token.text}" title="${token.title}" />`
+          break
+        case 'codespan':
+          str += `<code>${token.text}</code>`
+          break
+
+        default:
+          console.warn('Unknown inline token:', token.type)
+          break
+      }
+    }
+
+    return str
+  }
+
+  /** Processes markdown tokens recursively and incrementally builds the result. */
+  const processTokens = (tokens: Token[], parentDepth = 0) => {
+    // Stack of depths to determine indentation, scoped to the current parent
+    const stack: number[] = []
+
+    /** Helper function to get the indentation for a given depth. */
+    const indent = (depth: number) => '  '.repeat(depth + parentDepth)
+
+    for (const [index, token] of tokens.entries()) {
+      /** Helper function to check if ::/=scope is needed. */
```

Comment:
Suggested JSDOC:

/** Checks if the current token needs to be added to a special ::/=scope thought to separate it from its siblings. Needed for tables and ordered lists that are to be rendered at the same level with other siblings. **/